auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -43,53 +38,51 @@
 import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.auth.GoogleAuthProvider;
 
-public class SaveCredentialsActivity extends AppCompatBase
-        implements GoogleApiClient.ConnectionCallbacks, ResultCallback<Status>,
+import static android.app.Activity.RESULT_OK;
+
+public class SmartLock implements GoogleApiClient.ConnectionCallbacks, ResultCallback<Status>,
         GoogleApiClient.OnConnectionFailedListener {
     private static final String TAG = "CredentialsSaveBase";
     private static final int RC_SAVE = 100;
     private static final int RC_UPDATE_SERVICE = 28;
 
+    private AppCompatBase mActivity;
     private String mName;
     private String mEmail;
     private String mPassword;
     private String mProvider;
     private String mProfilePictureUri;
     private GoogleApiClient mCredentialsApiClient;
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.save_credentials_layout);
-
-        if (!FirebaseAuthWrapperFactory.getFirebaseAuthWrapper(mActivityHelper.getAppName())
-                .isPlayServicesAvailable(this)) {
-            finish(RESULT_FIRST_USER, getIntent());
-            return;
-        }
-
-        mName = getIntent().getStringExtra(ExtraConstants.EXTRA_NAME);
-        mEmail = getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
-        mPassword = getIntent().getStringExtra(ExtraConstants.EXTRA_PASSWORD);
-        mProvider = getIntent().getStringExtra(ExtraConstants.EXTRA_PROVIDER);
-        mProfilePictureUri = getIntent()
-                .getStringExtra(ExtraConstants.EXTRA_PROFILE_PICTURE_URI);
-
-        mCredentialsApiClient = new GoogleApiClient.Builder(this)
+    public SmartLock(AppCompatBase activity,
```

Comment:
If this class is going to be tied to an Activity, then one way to structure it would be a headless `Fragment`.  You'd call `SmartLock.getInstance(activity, ...)` which would attach a `Fragment` to the Activity.  Then that `Fragment` can actually launch activities for result and receive the results directly, rather than the constant `if (mSmartLock != null) { smartLock.onActivityResult(...) }`.